mm2src/coins/coin_balance.rs

```diff
@@ -482,6 +482,8 @@ pub mod common_impl {
             accounts: Vec::with_capacity(accounts.len() + 1),
         };
 
+        // FIXME: Here if this specific requested account was not found, we create it and `enable_hd_account` for it. But other accounts
+        //        are not taken into account. Note that this check in the preceding commit was `accounts.is_empty()`.
         if accounts.get(&path_to_address.account_id).is_none() {
```

Comment:
it seems this was introduced with EVM HD wallet PR, no idea why, maybe it was for trezor. Are there any drawback to using is_empty as it was before?

Reply:
I suppose it's an easy fix where we append this account to the list of accounts instead of dropping them.

Reply:
Would it be better to follow BIP-44 while fixing this? ref.

> Software should prevent a creation of an account if a previous account does not have a transaction history (meaning none of its addresses have been used before).
> 
> Software needs to discover all used accounts after importing the seed from an external source. Such an algorithm is described in "Account discovery" chapter.

Reply:
One more note about this line `if accounts.get(&path_to_address.account_id).is_none() {` , I suppose this block of code runs now whenever activating trezor ref. https://github.com/KomodoPlatform/komodo-defi-framework/blob/cdde4f169a76f40b06c1e2d8bfea89af752cefc3/mm2src/coins/eth/v2_activation.rs#L813 or am I missing something?

Reply:
> For HW HD wallets:
> - There is no way to derive an xpub without interacting with the hardware wallet. So we just don't load any accounts whatsoever.
> - On coin enables, the user will be prompted to derive the xpub for the account every time. i.e. next enable request will still ask them to derive the xpub nonetheless.

correct.

Reply:
> I suppose it's an easy fix where we append this account to the list of accounts instead of dropping them.

thx, done here: [c8282be](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2482/commits/c8282bebc9fe30724106f22a1c3bfd880c3264d3)